dora/src/semck/globaldef.rs

```diff
@@ -398,15 +459,40 @@ mod tests {
             pos(1, 15),
             SemError::ShadowStruct("Foo".into()),
         );
+        err(
+            "struct Foo {} struct Foo {}",
+            pos(1, 15),
+            SemError::ShadowStruct("Foo".into()),
+        );
         err(
             "struct Foo {} class Foo {}",
             pos(1, 15),
             SemError::ShadowStruct("Foo".into()),
         );
+        err(
+            "fun Foo() {} struct Foo {}",
+            pos(1, 14),
+            SemError::ShadowFunction("Foo".into()),
+        );
         err(
             "struct Foo {} fun Foo() {}",
             pos(1, 15),
-            SemError::ShadowStruct("Foo".into()),
+            SemError::ShadowStructConstructor("Foo".into()),
```

Comment:
Not sure we need ShadowStructConstructor right now, structs are not implemented yet but in the initial implementation would be intialized like this `Foo { .. }`. So from this we could see that this is a type and not a term. Wdyt?

Reply:
I would avoid having multiple different syntax variations that do the same thing (I would recommend that structs are initialized with `Foo(..)` and letting an IDE make the distinction between class and struct by coloring things differently.

Rust is a good example where giving initialization and calls different syntax has worked out quite poorly, I'd really try to avoid repeating those mistakes.

Reply:
Using `Foo(..)` would definitely make parsing easier. However we then really have to use named arguments at least in those cases: `Point(x: 1, y: 2)`.

Reply:
Agreed. I think named argument should not be special-case that works only within curly braces for structs like it does in Rust.

Instead, parameter lists should have named arguments -> constructors are parameter lists -> struct initializaters are constructors.

Reply:
Regarding named arguments: I would go with an intentionally constraint design:

- Params can be named or unnamed.
- Do not allow reordering of params.

This gets us 95% of the benefits and avoids many gnarly problems I experienced in the past.